backend/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java

```diff
@@ -794,4 +796,113 @@ public FutureProjectDAO(
             uac.getCollaboratorService().getResources(builder.build()), executor)
         .thenApply(GetResources.Response::getItemList, executor);
   }
+
+  public InternalFuture<List<GetResourcesResponseItem>> deleteProjects(List<String> projectIds) {
+    // validate argument
+    InternalFuture<Void> validateArgumentFuture =
+        InternalFuture.runAsync(
+            () -> {
+              // Request Parameter Validation
+              if (projectIds.isEmpty() || projectIds.stream().allMatch(String::isEmpty)) {
+                var errorMessage = "Project ID not found in request";
+                throw new InvalidArgumentException(errorMessage);
+              }
+            },
+            executor);
+
+    return validateArgumentFuture
+        .thenCompose(
+            unused -> {
+              // Get self allowed resources id where user has delete permission
+              return getSelfAllowedResources(
+                  ModelDBResourceEnum.ModelDBServiceResourceTypes.PROJECT,
+                  ModelDBActionEnum.ModelDBServiceActions.DELETE,
+                  projectIds);
+            },
+            executor)
+        .thenApply(
+            allowedProjectIds -> {
+              if (allowedProjectIds.isEmpty()) {
+                throw new PermissionDeniedException(
+                    "Delete Access Denied for given project Ids : " + projectIds);
+              }
+              return allowedProjectIds;
+            },
+            executor)
+        .thenCompose(
+            allowedProjectIds -> {
+              return getResourceItemsForWorkspace(
+                  "",
+                  Optional.of(allowedProjectIds),
+                  Optional.empty(),
+                  ModelDBResourceEnum.ModelDBServiceResourceTypes.PROJECT);
```

Comment:
Why do we need this? Don't we need just the IDs that we already have from `allowedProjectIds`?

Reply:
I need workspace ids for the `allowedProjectIds` so i have called again here and it will used while logging events. but yes might be i can simplify this in single uac call.